internal/transactionsubmission/engine/signing/signature_client.go

```diff
@@ -0,0 +1,101 @@
+package signing
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/stellar/go/txnbuild"
+	"golang.org/x/exp/slices"
+
+	"github.com/stellar/stellar-disbursement-platform-backend/db"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/transactionsubmission/engine/preconditions"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/transactionsubmission/utils"
+)
+
+//go:generate mockery --name=SignatureClient --case=underscore --structname=MockSignatureClient
+type SignatureClient interface {
+	NetworkPassphrase() string
+	SignStellarTransaction(ctx context.Context, stellarTx *txnbuild.Transaction, stellarAccounts ...string) (signedStellarTx *txnbuild.Transaction, err error)
+	SignFeeBumpStellarTransaction(ctx context.Context, feeBumpStellarTx *txnbuild.FeeBumpTransaction, stellarAccounts ...string) (signedFeeBumpStellarTx *txnbuild.FeeBumpTransaction, err error)
+	BatchInsert(ctx context.Context, amount int) (publicKeys []string, err error)
+	Delete(ctx context.Context, publicKey string) error
+	Type() string
```

Comment:
Why the type don't return the `SignatureClientType` here?

Reply:
This would cause a circular dependency with the auto-generated mocks :/